### ISSUE TITLE
ISPC build on ARM targeting x86 is broken

### ISSUE THREAD

**Issue by dbabokin** [ispc/ispc#2990]
```
[  3%] Generating share/ispc/builtins_dispatch.bc
cd /Users/dbabokin/ispc/dbabokin_ispc && /Users/dbabokin/llvm/bin-no-ncureses/bin/clang++ -x c -DREGULAR -fPIC -O2 -emit-llvm -c builtins/dispatch.c -o /Users/dbabokin/ispc/dbabokin_ispc/build/bin/../share/ispc/builtins_dispatch.bc
builtins/dispatch.c:33:36: error: invalid output constraint '=a' in asm
   33 |     __asm__ __volatile__("cpuid" : "=a"(info[0]), "=b"(info[1]), "=c"(info[2]), "=d"(info[3]) : "0"(infoType));
      |                                    ^
builtins/dispatch.c:37:36: error: invalid output constraint '=a' in asm
   37 |     __asm__ __volatile__("cpuid" : "=a"(info[0]), "=b"(info[1]), "=c"(info[2]), "=d"(info[3]) : "0"(level), "2"(count));
      |                                    ^
builtins/dispatch.c:45:53: error: invalid output constraint '=a' in asm
   45 |     __asm__ __volatile__(".byte 0x0f, 0x01, 0xd0" : "=a"(rEAX), "=d"(rEDX) : "c"(0));
      |                                                     ^
builtins/dispatch.c:57:53: error: invalid output constraint '=a' in asm
   57 |     __asm__ __volatile__(".byte 0x0f, 0x01, 0xd0" : "=a"(rEAX), "=d"(rEDX) : "c"(0));
      |                                                     ^
4 errors generated.
```

I'm macOS, but this should not be different on Linux or Windows. The problem is that `clang++` is run in "native" mode, i.e. the target is not specified. So x86 inline asm doesn't work. Any way, the dispatch is assumed to be x86 anyway.

Adding `--target=x86_64-apple-macosx` in my case fixes the problem. Though I'm not sure what generic x86 target we should use here to make it work on all platforms.

**Comment by nurmukhametov** [comment]
> I'm macOS, but this should not be different on Linux or Windows. The problem is that `clang++` is run in "native" mode, i.e. the target is not specified. So x86 inline asm doesn't work. Any way, the dispatch is assumed to be x86 anyway.

OK, so we need to explicitly specify `x86`.

> Adding `--target=x86_64-apple-macosx` in my case fixes the problem. Though I'm not sure what generic x86 target we should use here to make it work on all platforms.

I would like to avoid specifying here platform. What about just providing `-march=core2 -mtune=generic` instead, like here #3029  ?

**Comment by dbabokin** [comment]
Somehow `--target=x86_64-apple-macosx` works for me, but not `-march=core2 -mtune=generic`.

I hate how clang's target options work. They used to have "help" as value for `-march`, but not any more, only for `-mcpu`, which gives me only arm cpus.

I totally agree we need to have something generic. I look at this later today or tomorrow.

**Comment by dbabokin** [comment]
Clang requires to specify `--target` to enable `-march` and `-mtune` for non-host platform.

But dispatch module that we are building supposed to by universal for across platforms, right? At least before we converted it to a C file, we used single version of LLVM IR. I guess we should do the same with C version - i.e. compile it for Linux and verify that it works for Windows and macOS.

Also, if we don't pin the compilation to specific target+OS, the cross compiler will work differently than a native, which is a no go.
